### DISCUSSION TITLE
How to implement model into entity

### DISCUSSION THREAD

**Discussion by deanwean** [FabricMC/fabric-api#2006] [Mod Dev Support] [answered]
How do I add a model to my entity. I have already made the model, I just have no clue how to add the model. I have some of the entity stuff down, it's just the model script is so confusing.

**Comment by warjort** [comment, marked as answer]
Try following this:
https://fabricmc.net/wiki/tutorial:entity
For your specific question, look at the EntityRenderer part.

**Reply by deanwean** [reply]
So, I have to re-create my model in the script?

**Reply by warjort** [reply]
> So, I have to re-create my model in the script?

You don't really explain what you are talking about, so I will guess.
If you are using a custom external model format then you will need a 3rd party mod that handles it and follow its documentation.
Or translate it into java code.

Neither vanilla minecraft (java editiion) or the fabric api currently support custom entity models like it does for bedrock.
https://wiki.bedrock.dev/guide/custom-entity.html

Something is supposed to be added in minecraft 1.19?
see for example https://github.com/FabricMC/fabric/issues/1775

**Reply by deanwean** [reply]
I used BlockBench to create a model for my entity, I've got most of the entity code except for the model code. For me it seems all like gibberish to me since I'm completely new at this. Since I think I'm using a custom external model, how could I translate the model into java code? Or do you perhaps know any mods that could handle it?

**Reply by warjort** [reply]
BlockBench is not something I am familiar with.

I do know you can use external entity models with mods like:
https://www.curseforge.com/minecraft/mc-mods/geckolib
https://www.curseforge.com/minecraft/mc-mods/custom-entity-models-cem
or other mods like optifine

But how they work and whether they work with the blockbench format is something you will have to investigate with those mods or blockbench.

Manual translation into java would require finding some documentation for the block bench format and matching it to Mojang's entity model api, this is what those mods above are kind of doing.
